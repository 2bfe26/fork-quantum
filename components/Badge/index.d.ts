import React from 'react';

export interface BadgeProps {
    skin?: 'primary' | 'secondary' | 'success' | 'error' | 'neutral';
    children?: React.ReactNode[] | React.ReactNode;
    number?: number;
    inverted?: boolean;
    dot?: boolean;
    theme?: {
        baseFontSize?: number;
        spacing?: {};
        components?: {
            badge?: {};
        };
    };
}

export default class Badge extends React.Component<BadgeProps> {}
