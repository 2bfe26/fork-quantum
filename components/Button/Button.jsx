import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import SocialButtons from './SocialButtons';
import { SIZES } from '../Grid/sub-components/shared/grid-config';
import theme from '../shared/theme';
import skins from './skins';
import Icon from '../Icon/Icon';

const fontSize = ({ size }) => {
  const sizes = {
    xsmall: '12px',
    small: '12px',
    medium: '16px',
    big: '20px',
    xbig: '24px',
  };
  return `font-size: ${sizes[size] || sizes.medium};`;
};

const iconSize = ({ size }) => {
  const sizes = {
    normal: '24px',
  };

  return `font-size: ${sizes[size] || sizes.normal};`;
};

const iconMargin = ({ size }) => {
  const margins = {
    normal: '-5px 5px -7px -5px',
  };

  return `margin: ${margins[size] || margins.normal};`;
};

const padding = ({ size }) => {
  const paddings = {
    xsmall: '4px 12px',
    small: '8px 12px',
    medium: '10px 12px',
    big: '11px 16px',
    xbig: '13px 16px',
  };

  return `padding: ${paddings[size] || paddings.normal};`;
};

const ButtonIcon = styled(Icon)`
  ${iconMargin};
`;

const StyledButton = styled.button`
  border-radius: ${theme.sizes.radius};
  font-weight: bold;
  text-align: center;
  text-decoration: ${props => (props.link ? 'underline' : 'none')};

  ${fontSize}
  ${padding}
  ${theme.mixins.transition()};

  ${ButtonIcon} {
    ${iconSize}
  }

  ${props =>
    props.full &&
    `
    width: 100%;
  `}

  ${props =>
    props.center &&
    `
    display: block;
    margin-left: auto;
    margin-right: auto;
  `}

  ${props => `
    cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
  `}

  ${props => {
    const { unselected, selected, disabled, shadow } = skins(props);

    return `
      background-color: ${
        props.disabled ? disabled.background : unselected.background
      };
      border: 1px solid ${props.disabled ? disabled.border : unselected.border};
      color: ${props.disabled ? disabled.color : unselected.color};
      font-weight: ${
        props.disabled ? disabled.fontWeight : unselected.fontWeight
      };

      ${shadow ? theme.mixins.shadow() : undefined};

      &:active {
        ${shadow && theme.mixins.shadow(2)};
        background-color: ${selected.background};
        border-color: ${selected.border};
        color: ${selected.color};
      }
    `;
  }}

  @media (max-width: ${SIZES.phone}px) {
    margin-left: 0;
    margin-right: 0;
    width: 100%;
  }
`;

class Button extends React.Component {
  static Facebook = SocialButtons.Facebook;

  static Google = SocialButtons.Google;

  render() {
    const { children, icon, size, ...rest } = this.props;

    return (
      <StyledButton {...rest} size={size}>
        {icon && <ButtonIcon size={size} name={icon} />}
        {children}
      </StyledButton>
    );
  }
}

Button.defaultProps = {
  center: false,
  disabled: false,
  full: false,
  icon: '',
  size: 'medium',
  skin: 'primary',
  type: 'button',
  children: 'Catho',
  onClick: () => {},
};

Button.propTypes = {
  center: PropTypes.bool,
  disabled: PropTypes.bool,
  full: PropTypes.bool,
  /** Icon name. The full catalogue can be found
   * [here](/?selectedKind=1.%20Foundation&selectedStory=Icons) */
  icon: PropTypes.string,
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'big', 'xbig']),
  skin: PropTypes.oneOf(['primary', 'secondary', 'action']),
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
  children: PropTypes.node,
  onClick: PropTypes.func,
};

export default Button;
