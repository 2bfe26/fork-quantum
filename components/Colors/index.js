export default {
  PRIMARY: {
    50: '#E0EBF4',
    100: '#B3CEE3',
    200: '#80ADD1',
    300: '#4D8CBE',
    400: '#2674B0',
    500: '#005BA2',
    600: '#00539A',
    700: '#004990',
    800: '#004086',
    900: '#002f75',
  },
  SECONDARY: {
    50: '#E1F5FE',
    100: '#B3E5FC',
    200: '#81D4FA',
    300: '#4FC3F7',
    400: '#29B6F6',
    500: '#03A9F4',
    600: '#039BE5',
    700: '#0288D1',
    800: '#0277BD',
    900: '#01579B',
  },
  COMPLEMENTARY: {
    50: '#E3E3E8',
    100: '#BABAC5',
    200: '#8C8C9E',
    300: '#5D5E77',
    400: '#3B3C59',
    500: '#18193C',
    600: '#151636',
    700: '#11122E',
    800: '#0E0E27',
    900: '#08081A',
  },
  COMPLEMENTARY2: {
    50: '#E6E3E9',
    100: '#C2BAC8',
    200: '#998CA4',
    300: '#705E80',
    400: '#513C64',
    500: '#321949',
    600: '#2D1642',
    700: '#261239',
    800: '#1F0E31',
    900: '#130821',
  },
  ACTION: {
    50: '#F1F8E9',
    100: '#DCEDC8',
    200: '#C5E1A5',
    300: '#AED581',
    400: '#9CCC65',
    500: '#8BC34A',
    600: '#7CB342',
    700: '#689F38',
    800: '#558B2F',
    900: '#33691E',
  },
  WARMING: {
    50: '#FFF8E1',
    100: '#FFECB3',
    200: '#FFF082',
    300: '#FFD54F',
    400: '#FFCA28',
    500: '#FFC107',
    600: '#FFB300',
    700: '#FFA000',
    800: '#FF8F00',
    900: '#FF6F00',
  },
  DANGER: {
    50: '#FFEBEE',
    100: '#FFCDD2',
    200: '#EF9A9A',
    300: '#E57373',
    400: '#EF5350',
    500: '#F44336',
    600: '#E53935',
    700: '#D32F2F',
    800: '#C62828',
    900: '#B71C1C',
  },
  INFO: {
    50: '#E0F7FA',
    100: '#B2EBF2',
    200: '#80DEEA',
    300: '#4DD0E1',
    400: '#26C6DA',
    500: '#00BCD4',
    600: '#00ACC1',
    700: '#0097A7',
    800: '#00838F',
    900: '#006064',
  },
  P2P: {
    50: '#F8E0FD',
    100: '#EEB3FB',
    200: '#E380F8',
    300: '#D74DF5',
    400: '#CF26F3',
    500: '#C600F1',
    600: '#C000EF',
    700: '#B900ED',
    800: '#B100EB',
    900: '#A400E7',
  },
  GREY: {
    50: '#ECEFF1',
    100: '#CFD8DC',
    200: '#B0BEC5',
    300: '#90A4AE',
    400: '#78909C',
    500: '#647C8A',
    600: '#546E7A',
    700: '#455A64',
    800: '#37474F',
    900: '#263238',
  },
  BLACK: '#000000',
  WHITE: '#FFFFFF',
};
