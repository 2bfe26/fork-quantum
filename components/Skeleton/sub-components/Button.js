import styled from 'styled-components';
import { Rect } from './Rect';

export const Button = styled(Rect)`
  border-radius: 4px;

  height: ${({ size, theme: { spacing } }) => {
    const sizes = {
      xsmall: spacing.large,
      small: spacing.xlarge,
      medium: spacing.xxlarge,
      large: spacing.xxxlarge,
    };

    return `${sizes[size] || sizes.medium}px`;
  }};
`;
