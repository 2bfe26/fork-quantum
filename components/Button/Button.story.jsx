import React from 'react';
import { storiesOf } from '@storybook/react';
import HowToImport from '../../.storybook/decorators/HowToImport';
import Button from './Button';

const stories = storiesOf('Button', module);
stories
  .addDecorator(HowToImport)
  .add('Button', () => (
    <Button>Click me!</Button>
  ));
