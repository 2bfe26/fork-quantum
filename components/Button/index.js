import { Button, ButtonSimple } from './Button';

export { Button, ButtonSimple };
