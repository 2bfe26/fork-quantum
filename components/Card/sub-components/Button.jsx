import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {default as OriginalButton} from '../../Button';

const ButtonWrapper = styled(OriginalButton)`
  background-color: transparent;
  padding: 0;
  margin: 0 10px 0 0;
  box-shadow: none;
  font-family: oxygen;
  text-transform: uppercase;
  font-size: 12px;

  &:active {
    box-shadow: none;
    background-color: transparent;
    border: 1px solid transparent;
  }
`;

const Button = props => <ButtonWrapper {...props} />;

export default Button;