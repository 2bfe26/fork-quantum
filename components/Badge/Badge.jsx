import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Colors from '../Colors/deprecated';

const getBgColor = ({ skin }) => {
  const color = Colors[skin.toUpperCase()];
  return color[500] || color;
};

const getFontColor = ({ skin }) =>
  skin === 'white' ? Colors.BLACK : Colors.WHITE;

const Wrapper = styled.div`
  background-color: ${getBgColor};
  border-radius: 2px;
  color: ${getFontColor};
  font-size: 12px;
  padding: 2px 4px;
  text-align: center;
`;

const Badge = ({ children, skin, text }) => (
  <Wrapper skin={skin}> {children || text} </Wrapper>
);

Badge.propTypes = {
  skin: PropTypes.oneOf([
    'brand1',
    'brand2',
    'primary',
    'secondary',
    'success',
    'action',
    'warning',
    'danger',
    'info',
    'p2p',
    'black',
    'white',
  ]),
  children: PropTypes.string,
  text: PropTypes.string,
};

Badge.defaultProps = {
  skin: 'primary',
  children: '',
  text: 'Badge',
};

export default Badge;
