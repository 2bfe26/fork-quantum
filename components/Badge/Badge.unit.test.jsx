import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Badge from './Badge';
import Colors from '../Colors';
import Button from '../Button';
import 'jest-styled-components';

describe('<Badge />', () => {
  describe('Should match snapshot', () => {
    it('With number', () => {
      expect(renderer.create(<Badge number={10} />).toJSON()).toMatchSnapshot();
    });

    it('with children', () => {
      expect(
        renderer
          .create(
            <Badge number={10}>
              <Button />
            </Badge>,
          )
          .toJSON(),
      ).toMatchSnapshot();
    });
  });

  describe('Match correct styles', () => {
    const BadgeComponent = mount(<Badge number={10} />);

    it("should have correct colors when skin prop isn't defined", () => {
      expect(BadgeComponent.find('StyledBadge')).toHaveStyleRule(
        'background-color',
        Colors.BLACK[100],
      );

      expect(BadgeComponent.find('StyledBadge')).toHaveStyleRule(
        'color',
        Colors.BLACK[700],
      );
    });

    it("Should have spacing around when Badge don't have children", () => {
      expect(BadgeComponent.find('BadgeWrapper')).toHaveStyleRule(
        'margin-left',
        '8px',
      );
      expect(BadgeComponent.find('BadgeWrapper')).toHaveStyleRule(
        'margin-right',
        '8px',
      );
    });

    it('Should match colors when skin prop is defined', () => {
      const BadgeWithSkin = mount(<Badge number={10} skin="success" />);

      expect(BadgeWithSkin.find('StyledBadge')).toHaveStyleRule(
        'background-color',
        Colors.SUCCESS[200],
      );
      expect(BadgeWithSkin.find('StyledBadge')).toHaveStyleRule(
        'color',
        Colors.SUCCESS[900],
      );
    });
  });

  describe('Positioning when have children', () => {
    it('should be placed at top-right corner', () => {
      const BadgeComponent = mount(
        <Badge number={10}>
          <Button />
        </Badge>,
      );
      expect(BadgeComponent.find('BadgeWrapper')).toHaveStyleRule(
        'position',
        'relative',
      );
      expect(BadgeComponent.find('StyledBadge')).toHaveStyleRule(
        'position',
        'absolute',
      );
      expect(BadgeComponent.find('StyledBadge')).toHaveStyleRule(
        'top',
        '-10px',
      );
      expect(BadgeComponent.find('StyledBadge')).toHaveStyleRule(
        'right',
        '-10px',
      );
    });
  });
});
