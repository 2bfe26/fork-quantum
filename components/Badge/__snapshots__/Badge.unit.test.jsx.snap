// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> Should match snapshot With number 1`] = `
.c0 {
  display: inline-block;
  margin-left: 8px;
  margin-right: 8px;
}

.c1 {
  border-radius: 8px;
  display: inline-block;
  font-size: 12px;
  font-weight: bold;
  height: 20px;
  line-height: 20px;
  min-width: 20px;
  text-align: center;
  box-sizing: border-box;
  padding-left: 2px;
  padding-right: 2px;
  background-color: #f2f2f2;
  color: #4c4c4c;
}

<div
  className="c0"
  value={10}
>
  <span
    className="c1"
    value={10}
  >
    10
  </span>
  
</div>
`;

exports[`<Badge /> Should match snapshot with children 1`] = `
.c0 {
  display: inline-block;
  position: relative;
}

.c1 {
  border-radius: 8px;
  display: inline-block;
  font-size: 12px;
  font-weight: bold;
  height: 20px;
  line-height: 20px;
  min-width: 20px;
  text-align: center;
  box-sizing: border-box;
  padding-left: 2px;
  padding-right: 2px;
  position: absolute;
  right: -10px;
  top: -10px;
  background-color: #f2f2f2;
  color: #4c4c4c;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c2 *:nth-child(2) {
  margin-left: 5px;
}

.c2:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c2:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c2:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c2 .c3 {
  font-size: 24px;
}

<div
  className="c0"
  value={10}
>
  <span
    className="c1"
    value={10}
  >
    10
  </span>
  <button
    className="c2"
    disabled={false}
    onClick={[Function]}
    size="medium"
    type="button"
  >
    
  </button>
</div>
`;
